Fix DELETE query in derived delete operations

Huiyeongkim opened the pull request:
- Use correct ID column for child entities in JdbcDeleteQueryCreator
- Remove getIdDefiningParentPath() to reference current entity ID
- Rename parentIdColumns to idColumns for clarity
- Add test case for custom ID column name (@Column("CHILD_ID"))

Fixes errors when deleting child entities with non-standard ID column names.

Fixes #2123

<!--

Thank you for proposing a pull request. This template will guide you through the essential steps necessary for a pull request.
Make sure that:

-->

chanhyeong commented on spring-data-jdbc/src/test/java/org/springframework/data/jdbc/repository/JdbcRepositoryWithCollectionsChainHsqlIntegrationTests.java:
```diff
@@ -109,7 +110,7 @@ static class ChildElement {
 		String name;
 		Set<GrandChildElement> content = new HashSet<>();
-		@Id private Long id;
+        @Id @Column("CHILD_ID") private Long id;
```
I don't know how maintainers will think about this, but I was planning to maintain the existing codes, and create new test cases and classes to verify fixed.
The changes of test I made were just for reporting.

injae-kim replied:
> but I was planning to maintain the existing codes

Hi @chanhyeong! I'm helping @Huiyeongkim's first contribution by https://medium.com/opensource-contributors.

Can you explain more details about existing codes that you plan to edit?
We simply want to know how you think to fix this issue 🙇

Huiyeongkim replied:
@chanhyeong  
Thanks for pointing that out. Would it be okay if I try adding the test and update the PR accordingly?

chanhyeong replied:
Sure! I just wrote my opinion.

Huiyeongkim replied:
@chanhyeong 
Thank you! I’ve added the test as discussed.

schauder commented on spring-data-jdbc/src/test/java/org/springframework/data/jdbc/repository/JdbcRepositoryWithCollectionsCustomIdIntegrationTests.java:
```diff
@@ -0,0 +1,117 @@
+ */
+@IntegrationTest
+@EnabledOnDatabase(DatabaseType.HSQL)
+class JdbcRepositoryWithCollectionsCustomIdIntegrationTests {
```
I'd rather not have another full integration test class, but go with the approach you used in the reproducer of just tweaking the existing `JdbcRepositoryWithCollectionsChainHsqlIntegrationTests.java`

Huiyeongkim replied:
@schauder 
As suggested, I have updated JdbcRepositoryWithCollectionsChainHsqlIntegrationTests.java.
No new integration test class was added.

schauder reviewed (changes requested):
This PR causes `JdbcRepositoryEmbeddedWithCollectionIntegrationTests.deleteBy()` to fail. 

Please make sure to run all tests, if possible against all supported databases. 

You do this by running 

`./mvnw clean verify`
or 
`./mvnw clean verify -Pall-dbs`

The later requires Docker to run on your machine and also the acceptance of some db licenses, which you can accept by running `ci/accept-third-party-license.sh`. Of course, you should review any license you are accepting.

Huiyeongkim committed:
Fix DELETE queries with custom @Column ID names

- Support @Column annotation on ID fields when deleting collection entities
- Fallback to parent path IDs for embedded collections
- Add test for custom ID column DELETE operations

Fixes DATAJDBC-2123
Signed-off-by: Huiyeongkim <huiyeong9619@naver.com>
